railties/test/generators/app_generator_test.rb

```diff
@@ -383,6 +383,30 @@ def test_app_update_does_not_change_config_target_version
     assert_file "config/application.rb", /\s+config\.load_defaults 5\.1/
   end
 
+  def test_app_update_does_not_change_app_name_when_app_name_is_hyphenated_name
+    app_root = File.join(destination_root, "hyphenated-app")
+    run_generator [app_root, "-d", "postgresql"]
+
+    assert_file "#{app_root}/config/database.yml" do |content|
+      assert_match(/hyphenated_app_development/, content)
+      assert_no_match(/hyphenated-app_development/, content)
+    end
+
+    assert_file "#{app_root}/config/cable.yml" do |content|
+      assert_match(/hyphenated_app/, content)
+      assert_no_match(/hyphenated-app/, content)
+    end
+
+    FileUtils.cd(app_root) do
+      quietly { system("bin/rails app:update") }
```

Comment:
We should probably assert the return value of `Kernel#system`. ref. https://github.com/rails/rails/pull/32780#discussion_r186252217